src/features/inlay_hints.zig

```diff
@@ -188,29 +187,77 @@ fn writeBuiltinHint(builder: *Builder, parameters: []const Ast.Node.Index, argum
         const colonIndex = std.mem.indexOfScalar(u8, arg, ':');
         const type_expr: []const u8 = if (colonIndex) |index| arg[index + 1 ..] else &.{};
 
-        var label: ?[]const u8 = null;
+        var maybe_label: ?[]const u8 = null;
         var no_alias = false;
         var comp_time = false;
 
         var it = std.mem.splitScalar(u8, arg[0 .. colonIndex orelse arg.len], ' ');
         while (it.next()) |item| {
             if (item.len == 0) continue;
-            label = item;
+            maybe_label = item;
 
             no_alias = no_alias or std.mem.eql(u8, item, "noalias");
             comp_time = comp_time or std.mem.eql(u8, item, "comptime");
         }
 
+        const label = maybe_label orelse return;
+        if (label.len == 0 or std.mem.eql(u8, label, "...")) return;
+
         try builder.appendParameterHint(
             tree.firstToken(parameter),
-            label orelse "",
+            label,
             std.mem.trim(u8, type_expr, " \t\n"),
             no_alias,
             comp_time,
         );
     }
 }
 
+/// Takes a variable declaration AST node. If the type is inferred, attempt to infer it and display it as a hint.
+fn writeVariableDeclHint(builder: *Builder, decl_node: Ast.Node.Index) !void {
+    const tracy_zone = tracy.trace(@src());
+    defer tracy_zone.end();
+
+    const handle = builder.handle;
+    const tree = handle.tree;
+
+    const hint = tree.fullVarDecl(decl_node) orelse return;
+    if (hint.ast.type_node != 0) return;
+
+    const resolved_type = try builder.analyser.resolveTypeOfNode(.{ .handle = handle, .node = decl_node }) orelse return;
+
+    var type_references = Analyser.ReferencedType.Set.init(builder.arena);
+    var reference_collector = Analyser.ReferencedType.Collector.init(&type_references);
+
+    var type_str: []const u8 = "";
+    try builder.analyser.referencedTypes(
+        resolved_type,
+        &type_str,
+        &reference_collector,
+    );
+    if (type_str.len == 0) return;
+
+    const assign_token = tree.firstToken(hint.ast.init_node) - 1;
+
+    const decl_str = offsets.tokensToSlice(tree, tree.firstToken(hint.ast.init_node) - 2, assign_token);
+    var decl_spaces: usize = 0;
+    for (decl_str) |char| {
+        if (char == ' ') {
+            decl_spaces += 1;
+        }
+    }
+
+    try builder.hints.append(builder.arena, .{
+        .index = offsets.tokenToIndex(tree, assign_token) - decl_spaces,
```

Comment:
I should have looked more closely at your code that finds the inlay hint index because you can get the end of a token in the source with `offsets.tokenToLoc(tree, token_index).end`. `hint.ast.mut_token + 1` is always the variable name token.
```suggestion
    try builder.hints.append(builder.arena, .{
        .index = offsets.tokenToLoc(tree, hint.ast.mut_token + 1).end,
```